.circleci/config.yml

```diff
@@ -23,23 +23,45 @@ jobs:
       - run:
           name: Install dependencies
           command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r .circleci/requirements.txt
+            sudo pip install -r .circleci/requirements.txt
 
       - save_cache:
-          paths:
-            - ./venv
           key: v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
+          paths:
+            - ~/.cache 
 
       - run:
-          name: Linting
+          name: Run linter
           command: |
-            . venv/bin/activate
-            flake8 --max-line-length 160 --per-file-ignores='datascroller/__init__.py:F401' datascroller/
+            # stop the build if there are Python syntax errors or undefined names
+            flake8 datascroller --count --select=E901,E999,F821,F822,F823 \
+              --show-source --statistics
+            # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide      
+            flake8 datascroller --count --exit-zero --max-complexity=10 \
```

Comment:
What does the `max-complexity` flag do?

Reply:
Glad you dug into this!

Reply:
@johncmerfeld, reading about this now. A guy named Thomas McCabe came up with a software metric to indicate the complexity of a program in 1976. It is called [Cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and counts linearly independent paths through a programs source code. According to McCabe, anything beyond 10 is too complex.

What do you think? Leave it in or pull it?

Reply:
Cool concept. I guess if we find ourselves bumping into this limit we'll want to look into it. No harm in leaving it in!